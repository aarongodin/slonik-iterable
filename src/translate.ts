import { sql, ValueExpressionType } from "slonik";

export function translateValue(
  column: string,
  value: any
): ValueExpressionType {
  let currentValue = value;

  if (Array.isArray(currentValue)) {
    currentValue = sql.array(currentValue, "text");
  }

  return currentValue;
}
