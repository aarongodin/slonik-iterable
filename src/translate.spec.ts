import { translateValue } from "./translate"

describe("translate", () => {
  describe("translateValue()", () => {
    it("returns the current value given the value is not a special type", () => {
      const actual = translateValue("myCol", "just a string")
      expect(actual).toEqual("just a string")
    })

    it("returns a slonik array for values that are arrays", () => {
      const actual = translateValue("myCol", ["is", "an", "array!"])
      expect(actual).toMatchSnapshot()
    })
  })
})
