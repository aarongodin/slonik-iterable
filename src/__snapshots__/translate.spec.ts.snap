// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translate translateValue() returns a slonik array for values that are arrays 1`] = `
Object {
  "memberType": "text",
  "type": "SLONIK_TOKEN_ARRAY",
  "values": Array [
    "is",
    "an",
    "array!",
  ],
}
`;
